### PR TITLE
Add Pangeo Binder button

### PR DESCRIPTION
**Pull request by weiji14**
Specifying PyGMT dependencies in a conda environment.yml file. Double check if there are any other dependencies we need?

Test this branch at [Binder](https://binder.pangeo.io/v2/gh/GenericMappingTools/scipy2021/binder)

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+name: pygmt-scipy2021
+channels:
+  - conda-forge
+dependencies:
+  - conda-forge::gmt=6.1.1
+  - conda-forge::jupyterlab=3.0
```
Comment (liamtoney) [inline environment.yml:6]: I assume `jupyterlab` installs the traditional `notebook` too as a dep?

File: environment.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+name: pygmt-scipy2021
+channels:
+  - conda-forge
+dependencies:
+  - conda-forge::gmt=6.1.1
```
Comment (liamtoney) [inline environment.yml:5]: The `conda-forge::` syntax is new to me. Needed if we specify the channel above?

**Review by liamtoney — COMMENTED** [review]
I left a few questions on the `.yml` file. Using Binder makes a lot of sense!

**Comment by liamtoney** [comment]
Okay, so after launching Binder I see that we have
```
jovyan@jupyter-genericmappingtools-2dscipy2021-2dkqh2ipxw:~$ conda env list
# conda environments:
#
base                     /srv/conda
notebook              *  /srv/conda/envs/notebook
```
with `notebook` containing the deps listed in the `.yml` file. Sweet.

File: environment.yml (hunk 2 above)
Comment (weiji14) [reply environment.yml:5]: Not actually needed since they're all conda-forge. But keeping it doesn't hurt either.

File: environment.yml (hunk 1 above)
Comment (weiji14) [reply environment.yml:6]: I think so, even with jupyterlab 3.0? There's a single view option that looks pretty much like a classic notebook.

**Pull request merged**